### PR TITLE
ADBDEV-4959: Fix segfault when copying dropped column ACLs to a new partition.

### PR DESCRIPTION
**Pull request by RekGRpth**
Fix segfault when copying dropped column ACLs to a new partition.

The CopyRelationAcls function copies ACLs without paying attention to dropped
columns. This can lead to a segfault when adding a new partition. This patch
skips dropped columns when copying ACLs.

### CODE REVIEW

**Comment by BenderArenadata** [comment]
Allure report https://allure-ee.adsw.io/launch/65352

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1103017

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1103018

**Review by dnskvlnk — APPROVED** [review]

File: src/backend/catalog/aclchk.c [hunk 1]
```diff
@@ -2321,6 +2321,9 @@ CopyRelationAcls(Oid srcId, Oid destId)
 		bool		nulls[Natts_pg_attribute];
 		bool		replaces[Natts_pg_attribute];
 
+		if (attSrcForm->attisdropped)
+			continue;
+
```
Comment (KnightMurloc) [inline src/backend/catalog/aclchk.c:2326]: maybe combine the check with the isNull below?

Comment (RekGRpth) [reply src/backend/catalog/aclchk.c:2326]: Why make unnecessary calls to SysCacheGetAttr?

**Review by KnightMurloc — APPROVED** [review]

**Comment by BenderArenadata** [comment]
Allure report https://allure-ee.adsw.io/launch/65966

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1128943

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1128944

**Comment by BenderArenadata** [comment]
Failed job Regression tests with ORCA on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1128938

**Comment by BenderArenadata** [comment]
Failed job Regression tests with ORCA on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1131970

**Comment by BenderArenadata** [comment]
Failed job Regression tests with ORCA on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1132636

**Pull request merged**